### PR TITLE
feat(favicon): add a check for custom favicon

### LINKED ISSUE #261: Feature request custom favicon
https://github.com/webpro/reveal-md/blob/6d068644d2ee5bfdecb85f6e3f45f8b35bf44b19/lib/server.js#L22

- load `favicon.ico` from same directory of markdown file

```
reval-md foo/bar.md <- should check for foo/favicon.ico, overwise load lib/favicon.ico

### PR DESCRIPTION
**Pull request by zladuric**
Hi, 
I'd like to resolve #261 here, the feature would be useful to me too. I've ran test and lint on windows, have nothing else handy right now.

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -19,7 +27,9 @@ module.exports = async () => {
   const initialDir = await getInitialDir();
   const initialPath = await getInitialPath();
 
-  app.use(favicon(path.join(__dirname, 'favicon.ico')));
+  const customFavicon = await getCustomFavicon();
+  const faviconPath = customFavicon || path.join(__dirname, 'favicon.ico');
```
Comment (webpro) [inline lib/server.js]: There is logic for the favicon in three locations (here, util, and config). Could you please put it all in one location please? I guess the config module makes sense.

File: lib/config.js [hunk 2]
```diff
@@ -82,6 +82,20 @@ module.exports.getTemplate = async template => {
...
+  const faviconPath = path.join(initialDir, 'favicon.ico');
+  const faviconExists = await fs.pathExists(faviconPath);
+  if (!faviconExists) {
+    return null;
+  }
+  const isFaviconFile = await isFile(faviconPath);
+  if (!isFaviconFile) {
+    return null;
+  }
+  return faviconPath;
```
Comment (webpro) [inline lib/config.js]: How about simplifying this a bit, something like:

```
return await isFile(faviconPath) ? faviconPath : null;
```

File: lib/server.js (hunk 1 above)
Comment (zladuric) [reply lib/server.js]: I've moved the favicon path resolution to config.js. I initially placed the default one to server, as in the case of moving `config.js` elsewhere then `server.js` would then break the default favicon behavior. I guess premature optimization or something. Anyway I've moved the logic to config now, and I've also added a line to docs (README) to document this for reveal-md users.

**Comment by webpro** [comment]
Thanks, @zladuric! It's released in [v3.1.0](https://github.com/webpro/reveal-md/releases/tag/3.1.0)

**Pull request merged**